React_assignments/react_2nd_module/todolist_practise/src/components/TodoApp.jsx

```diff
@@ -0,0 +1,74 @@
+import { AddTodoForm } from "./AddTodoForm";
+import { TodoFilter } from "./TodoFilter";
+import { TodoList } from "./TodoList";
+import { useCallback, useEffect, useState } from "react";
+import { useFetch } from "./useFetch";
+
+export const TodoApp = () => {
+  const [showCompleted, setShowCompleted] = useState(false);
+  const [addTodoInput, setAddTodoInput] = useState("");
+
+  const getTodos = useCallback(async () => {
+    const response = await fetch("https://jsonplaceholder.typicode.com/todos");
+
+    if (!response.ok) {
+      throw new Error("Failed to fetch todos");
+    }
+    return response.json();
+  }, []);
+
+  const { todos, loading, error } = useFetch(getTodos);
+  const [fetchedTodo, setFetchedTodo] = useState([]);
+
+  useEffect(() => {
+    if (todos) {
+      setFetchedTodo(todos);
+    } else return;
+  }, [todos]);
+
+  const addTodo = (title) => {
+    setFetchedTodo([
+      ...fetchedTodo,
+      { id: fetchedTodo.length + 1, title, completed: false },
+    ]);
+    setAddTodoInput("");
+  };
+
+  const markTodoCompleted = (id, completed) => {
+    console.log(id, completed);
+    setFetchedTodo(
+      fetchedTodo.map((todo) => {
```

Comment:
You can get the all the value into seperate array and then set the setFetchedtodo